pkg/networkmanager/interfaces.js

```diff
@@ -3256,6 +3243,32 @@ function apply_master_slave(choices, model, apply_master, master_connection, mas
         return cockpit.all(deferreds.get());
     }
 
+    if (!master_connection) {
+        var active_settings = [ ];
+
+        if (master_settings.bond &&
+            master_settings.bond.options &&
+            master_settings.bond.options.primary) {
+            var iface = model.find_interface(master_settings.bond.options.primary);
```

Comment:
I would suggest defining ```iface``` at the top level of this function. Otherwise in a long function javascript variable hoisting becomes confusing.